### PR TITLE
Add --font-weight-bold and set previous bold to 601

### LINKED ISSUE #24305: Bold text not rendering for some fonts at font-weight: 600 e.g. 'Segoe UI'
### Description

I noticed this in my hosted 1.18.5 instance: bold wiki text was not showing as bold in latest Chrome browser.  I went straight to try.gitea.io and reproduced the issue there on the current dev instance.  Curiously, this issue does not present with Firefox, which renders the bold as expected, so it seems Chrome is rendering the same CSS differently [EDIT: more likely different fonts, as the Chrome instance was on Linux and the Firefox test was on a windows 10 box].

Video showing my experience:

https://user-images.githubusercontent.com/1931303/234031441-3ae06ce1-cd63-4742-8cbf-24c7d0ec1f60.mp4

There seems to be two CSS definitions for STRONG that are related to this issue.  When I disable one of them, as shown in the video, bold returns.

### Gitea Version

try.gitea.io

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

1.20.0+dev-559-g75c62054a

### Operating System

Linux (Debian)

### How are you running Gitea?

try.gitea.io

### Database

None

### PR DESCRIPTION
**Pull request by wxiaoguang**
Fix #24305

According to MDN, "bold" starts from 700, some fonts do not provide "bolding" for weight 600

https://developer.mozilla.org/en-US/docs/Web/CSS/font-weight

### CODE REVIEW

**Comment by wxiaoguang** [comment]
Would you take this PR? I am open.

**Comment by wxiaoguang** [comment]
As according to https://github.com/go-gitea/gitea/issues/24305#issuecomment-1520366666, maybe we can use 610 ?

**Review by delvh — APPROVED** [review]

**Comment by silverwind** [comment]
Let's wait on response for https://github.com/go-gitea/gitea/issues/24305#issuecomment-1520392895. Maybe we need to do some kind of detection and if a semi-bold is available, use that. `700` or `bold` seems too bold for my tastes.

**Comment by silverwind** [comment]
> Would you take this PR? I am open (I am not sure about how to satisfy all users, but some fonts don't work with 600)
> 
> Actually `bold` seems safer for all fonts and browsers.

You can put the variable, we can then tune later.

**Comment by wxiaoguang** [comment]
> You can put the variable, we can then tune later.

Done, feel free to edit and fine tune.

**Comment by silverwind** [comment]
So, `bold` is an alias to `700`. I think there is no way around using it to ensure all platforms can show distinct `700` vs `400, and I think CSS unfortunately offers no way to fine-tune for platforms that support intermediate thickness inbetween, e.g. I would prefer `600` if it is available and distinct from `700`.

**Comment by wxiaoguang** [comment]
In the latest commit I used 610, what do you think about it? I guess it could satisfy most users.

**Comment by silverwind** [comment]
Ah, I see, so if `610` works, it does sound like a good compromise. Results in my machine:

Screenshot 2023-04-24 at 17 43 49

**Comment by wxiaoguang** [comment]
> It does look like `610` renders more like `500` than `600` for me.

Noooo .... because the `1` is different from `0`, that's not mono-font.

610 is definitely slightly larger than 600

**Comment by silverwind** [comment]
Ah, yes, it's a visual deception :D

**Review by silverwind — APPROVED** [review]

**Comment by silverwind** [comment]
FTR, I see no difference between 601 and 610, so we may be able to reduce to 601 for the same effect. Any reason why specifically 610?

Screenshot 2023-04-24 at 18 01 23

**Comment by wxiaoguang** [comment]
No particular reason indeed ...  Personally I like to leave some buffer space when writing code 😁 

It could be lowed to 601 if you like.

File: web_src/css/base.css [hunk 1]
```diff
@@ -3,6 +3,9 @@
   --fonts-proportional: -apple-system, "Segoe UI", system-ui, "Roboto", "Helvetica Neue", "Arial";
   --fonts-monospace: "SFMono-Regular", "Menlo", "Monaco", "Consolas", "Liberation Mono", "Courier New", monospace, var(--fonts-emoji);
   --fonts-emoji: "Apple Color Emoji", "Segoe UI Emoji", "Noto Color Emoji", "Twemoji Mozilla";
+  /* "font-weight: bold" starts from 700, some fonts do not provide "bolding" for weight 600.
+  But some users consider "700" is too heavy, so use 610 here to bypass some font's rule, eg: "Segoe UI" starts bolding from 601. */
+  --font-weight-bold: 610;
```
Comment (silverwind) [inline web_src/css/base.css]:
```suggestion
  /* "font-weight: bold" starts from 700, some fonts do not provide "bolding" for weight 600. */
  /* But some users consider "700" is too heavy, so use 601, which is when Segoe UI on Linux */
  /* starts bolding. */
  --font-weight-bold: 601;
```

**Comment by silverwind** [comment]
Yeah I get it, but reading MDN indicates that browsers are free to scale the font how they like so it's a fluid 1-1000, even thought most browsers limit to steps of 100, the spec allows and encourages step size 1 😉.

**Comment by silverwind** [comment]
Actually I think I will do one more small tweak here, introducing a semibold weight as well.

**Comment by silverwind** [comment]
`--font-weight-semibold` added and applied to dashboard repolist. I think it looks cleaner this way:

Before (601):

Screenshot 2023-04-24 at 18 56 46

After (500):

image

**Comment by wxiaoguang** [comment]
> `--font-weight-semibold` added and applied to repolist. I think it looks cleaner this way.

I guess we do not need to bold them, just use normal font?

**Comment by silverwind** [comment]
Right, that's fine as well.

**Pull request merged**